pkg/proxy/iptables/proxier.go

```diff
@@ -1376,15 +1376,36 @@ func (proxier *Proxier) syncProxyRules() {
 		}
 
 		// Now write loadbalancing & DNAT rules.
-		n := len(endpointChains)
-		localEndpointChains := make([]utiliptables.Chain, 0)
+
+		// Firstly categorize each endpoint into three buckets:
+		//   1. all endpoints that are ready and NOT terminating.
+		//   2. all endpoints that are local, ready (NOT terminating).
+		//   3. all endpoints that are local, serving and terminating.
+		readyEndpointChains := make([]utiliptables.Chain, 0)
```

Comment:
if you're going to explicitly make the arrays here, estimate the size, which is bounded by `len(endpointChains)` but could probably be 1/3 or 1/2.  `make([]utiliptables.Chain, 0)` is not kube style and doesn't offer any benefit.  I don't think we're estimating zero size and i don't think there's any value in that (if you think something won't be allocated or will be o(0-5), ALWAYS use `var x []*endpointsInfo` because then you avoid the heap allocation when you don't need it)

Reply:
yup, good catch, wil update